driver-core/src/main/com/mongodb/internal/connection/InternalStreamConnection.java

```diff
@@ -607,8 +603,14 @@ public void failed(final Throwable t) {
                 }
             });
         } catch (Throwable t) {
-            close();
-            callback.onResult(null, t);
+            try (AutoCloseable second = () -> callback.onResult(null, t);
+                 AutoCloseable first = this::close) {
+                releaseAllBuffers(byteBuffers);
```

Comment:
I don't think this is correct.  The callback that is passed to this method already has responsibility for releasing the buffers.   See line 490 in this file.

Reply:
Thanks you! 👏  It is great that you spotted this. Rolled back the changes to this method.